server/src/main/java/org/opensearch/common/remote/AbstractRemoteWritableBlobEntity.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.remote;
+
+import org.opensearch.common.blobstore.BlobPath;
+import org.opensearch.core.compress.Compressor;
+import org.opensearch.core.xcontent.NamedXContentRegistry;
+import org.opensearch.gateway.remote.ClusterMetadataManifest.UploadedMetadata;
+
+import static org.opensearch.gateway.remote.RemoteClusterStateUtils.PATH_DELIMITER;
+
+/**
+ * An extension of {@link RemoteWriteableEntity} class which caters to the use case of writing to and reading from a blob storage
+ *
+ * @param <T> The class type which can be uploaded to or downloaded from a blob storage.
+ */
+public abstract class AbstractRemoteWritableBlobEntity<T> implements RemoteWriteableEntity<T> {
+
+    protected String blobFileName;
+
+    protected String blobName;
+    private final String clusterUUID;
+    private final Compressor compressor;
+    private final NamedXContentRegistry namedXContentRegistry;
+
+    public AbstractRemoteWritableBlobEntity(
+        final String clusterUUID,
+        final Compressor compressor,
+        final NamedXContentRegistry namedXContentRegistry
+    ) {
+        this.clusterUUID = clusterUUID;
+        this.compressor = compressor;
+        this.namedXContentRegistry = namedXContentRegistry;
+    }
+
+    public abstract BlobPathParameters getBlobPathParameters();
+
+    public String getFullBlobName() {
+        return blobName;
+    }
+
+    public String getBlobFileName() {
+        if (blobFileName == null) {
+            if (blobName == null) {
+                return null;
+            }
+            String[] pathTokens = blobName.split(PATH_DELIMITER);
```

Comment:
you can cache the pathTokens once generated in case it is being accessed multiple times. can be TODO for now and you can expose a method to get pathTokens directly as well.